numba/cpython/builtins.py

```diff
@@ -73,6 +73,20 @@ def opaque_is(context, builder, sig, args):
         return cgutils.false_bit
 
 
+@lower_builtin(operator.is_, types.Boolean, types.Boolean)
+def bool_is_impl(context, builder, sig, args):
+    arg1, arg2 = args
+    arg1_type, arg2_type = sig.args
+    _arg1 = context.cast(builder, arg1, arg1_type, types.boolean)
+    _arg2 = context.cast(builder, arg2, arg2_type, types.boolean)
+    eq_impl = context.get_function(
+        operator.eq,
+        typing.signature(types.boolean, types.boolean, types.boolean)
+    )
+    return eq_impl(builder, (_arg1, _arg2))
```

Comment:
Why is this necessary? Doesn't `generic_is` above handle it?

Reply:
@eric-wieser This is to support cases when one arg is BooleanLiteral and other is types.boolean. In this case `generic_is` always returns False since `lhs_type == rhs_type` check is not met.

Reply:
Two points would be good to include in comments here:
* This covers both Boolean and BooleanLiteral types
* This is needed for books because bools in python are guaranteed to preserve identity. It's not needed for any other literal types.